pymodbus/logging.py

```diff
@@ -32,12 +32,16 @@ class Log:
     _logger = logging.getLogger(__name__)
 
     @classmethod
-    def apply_logging_config(cls, level=logging.WARNING):
-        """Apply basic logging configuration"""
-        logging.basicConfig(
-            format="%(asctime)s %(levelname)-5s %(module)s:%(lineno)s %(message)s",
-            datefmt="%H:%M:%S",
-        )
+    def apply_logging_config(
+        cls, level: int = logging.WARNING, **config_overrides: dict
+    ) -> None:
+        """Apply basic logging configuration."""
+        default_config: dict = {
+            "format": "%(asctime)s %(levelname)-5s %(module)s:%(lineno)s %(message)s",
+            "datefmt": "%H:%M:%S",
+        }
+        config_kwargs = {**default_config, **config_overrides}
```

Comment:
This still allows to overwrite DEBUG. You are actually a good example, if you overwrite DEBUG, you are not able to submit issues, without somehow changing the overwrite.

In your case it would be a lot simpler, not to call our function, but do
```
  pymodbus.Log.LOG_LEVEL = <what you want>
```
A small change in the Log class could actually avoid that, by moving that line into build_msg.

have LOG_LEVEL = None 
and do:
```
   if not cls.LOG_LEVEL:
       cls.LOG_LOVEL = cls…….
```

IF you do not call our function your basic_config will rule.

Reply:
> ```python
>    if not cls.LOG_LEVEL:
>        cls.LOG_LOVEL = cls…….
> ```

I don't quite understand this part.  What does the `None` sentinel level mean?

Does this go into `build_msg` or `apply_logging_config`, and where within that method?  Why is it mutating the `Log.LOG_LEVEL` if `cls.LOG_LEVEL = None`?

Reply:
if you do not call apply_logging_config, cls.LOG_LEVEL still need to be set.

cls.LOG_LEVEL starts of being None, to secure it is set to effective log level, first time build_msg is called.

Reply:
I think `cls.LOG_LEVEL` will fail because `None` comparisons in checks like this `logging.INFO >= cls.LOG_LEVEL` will fail as a `TypeError` before we even get to `build_msg`.

Reply:
no it will not.

Reply:
```python
>>> logging.INFO >= None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>=' not supported between instances of 'int' and 'NoneType'
```

I don't understand what you mean, I am sorry.